server/src/pages/shotindex/controller.js

```diff
@@ -25,33 +25,51 @@ exports.launch = function(m) {
     }
     Object.assign(m, extractQueryParamValues(queryParamModelPropertyMap));
     model = m;
-    render();
-    if (firstRun && m.shots === null) {
-      // The actual shot data hasn't been loaded yet, so we'll immediately request it:
-      refreshModel();
-    }
-    firstRun = false;
-    return;
   }
   if (window.wantsauth) {
-    if (window.wantsauth.getAuthData() && !location.search.includes("reloaded")) {
-      location.search = "reloaded";
+    // Handle non-owner my shots page redirection
+    if (!m.hasDeviceId) {
+      if (window.wantsauth.getAuthData() && !location.search.includes("reloaded")) {
+        location.search = "reloaded";
+        return;
+      }
+      const authTimeout = setTimeout(() => {
+        // eslint-disable-next-line no-global-assign, no-native-reassign
+        location = location.origin + "/#tour";
+      }, FIVE_SECONDS);
+      window.wantsauth.addAuthDataListener((data) => {
+        if (location.search.indexOf("reloaded") > -1) {
+          return;
+        }
+        clearTimeout(authTimeout);
+        location.search = "reloaded";
+      });
       return;
     }
-    const authTimeout = setTimeout(() => {
-      // eslint-disable-next-line no-global-assign, no-native-reassign
-      location = location.origin + "/#tour";
-    }, FIVE_SECONDS);
-    window.wantsauth.addAuthDataListener((data) => {
-      if (location.search.indexOf("reloaded") > -1) {
-        return;
+
+    if (window.wantsauth.getAuthData()) {
+      Object.assign(model, window.wantsauth.getAuthData());
+      if (firstRun && m.shots === null) {
+        // The actual shot data hasn't been loaded yet, so we'll immediately request it:
+        refreshModel();
       }
-      clearTimeout(authTimeout);
-      location.search = "reloaded";
-    });
+      firstRun = false;
+    } else {
+      window.wantsauth.addAuthDataListener((data) => {
+        Object.assign(model, data);
+        if (firstRun && m.shots === null) {
+          // The actual shot data hasn't been loaded yet, so we'll immediately request it:
+          refreshModel();
+        }
+        firstRun = false;
+        render();
+      });
```

Comment:
By the time `refreshModel()` is called, the cookies should be available.  Otherwise the `reloaded` code would've executed.  So I don't think we need wrap `refreshModel` inside wantsauth.

Reply:
@chenba  reloaded code relies on deviceId which can return true if deviceId cookie is set and accountId is not ( before confirm-login redirection from Firefox Account authentication)

Is the ask above to remove refreshModel from addAuthDataListener or completely out of window.wantsauth as it exists right now in master

Reply:
Updated code to move refreshModel out of window.wantsauth. The code is still listening to wantsauth data listener and re-rendering code on authinfo retrieval